Yosemite/Yosemite/Stores/AppSettingsStore.swift

```diff
@@ -467,6 +485,62 @@ private extension AppSettingsStore {
     }
 }
 
+// MARK: - Products Settings
+//
+private extension AppSettingsStore {
+    func loadProductsSettings(siteID: Int64, onCompletion: (Result<StoredProductSettings, Error>) -> Void) {
+        guard let allSavedSettings: [StoredProductSettings] = try? fileStorage.data(for: productsSettingsURL) else {
+            let error = AppSettingsStoreErrors.noProductsSettings
+            onCompletion(.failure(error))
+            return
+        }
+
+        let settings = allSavedSettings.filter { $0.siteID == siteID }
+
+        guard let settingsUnwrapped = settings.first else {
+            let error = AppSettingsStoreErrors.noProductsSettings
+            onCompletion(.failure(error))
+            return
+        }
```

Comment:
⚠️ Have we considered encapsulating the uniqueness of `siteID` inside the `struct` instead? Something like:

```swift
/// Dummy name
public struct ProductsModuleSettings: Codable, Equatable {

    struct Filter {
        public let siteID: Int64
        public let sort: String?
        public let stockStatusFilter: ProductStockStatus?
        public let productStatusFilter: ProductStatus?
        public let productTypeFilter: ProductType?
    }

    /// SiteID: Filter
    let filters: [Int64: Filter]
}
```

In this way, we can hide this complexity as methods inside the `ProductsModuleSettings`. An example are these methods inside `GeneralAppSettings`:

https://github.com/woocommerce/woocommerce-ios/blob/42402d6b6799e24eb5b5d832476cb712fe2f7f2d/Storage/Storage/Model/GeneralAppSettings.swift#L28-L46

If we do this, I hope this class will be a bit shorter and will mostly only contain file saving operations.

Reply:
The class will be just 2 or 3 lines shorter, but at the same time we should introduce a new Codable model 😄 however if you think we can really get a benefit from it, it doesn't cost me anything to carry out this migration (also because once the model is defined, it would be difficult to modify it later)

Reply:
Yeah, if you're fine with that, let's do it. I think these parts could all be encapsulated in the `struct`: 

https://github.com/woocommerce/woocommerce-ios/blob/46b7fd986c225f0d05912fede77dca5f5a6b4c58/Yosemite/Yosemite/Stores/AppSettingsStore.swift#L498-L500

https://github.com/woocommerce/woocommerce-ios/blob/46b7fd986c225f0d05912fede77dca5f5a6b4c58/Yosemite/Yosemite/Stores/AppSettingsStore.swift#L517

https://github.com/woocommerce/woocommerce-ios/blob/46b7fd986c225f0d05912fede77dca5f5a6b4c58/Yosemite/Yosemite/Stores/AppSettingsStore.swift#L520-L525

The benefits that I see are:

- `ProductsModuleSettings` encapsulates loading `Filter` settings by `siteID` or updating by the `Filter` settings by `siteID`. It's the only object that knows how to update itself. 
- `AppSettingsStore` is only concerned about saving and loading the `plist` that contains the `ProductsModuleSettings`. 

These are significant to me even though we'd only be saving a few lines.

Reply:
Thank you @shiki for the suggestions. I made the changes and you can find them in this commit 95be4c7048ea72f7b7f0051652a5f113886a9ad2
Consider that I decided to not use the word "filter` since we want to use them in a generic way, so the struct inside is called `Setting` :D